src/Control/Inciter/InputDeck/Grammar.hpp

```diff
@@ -229,6 +229,15 @@ namespace grm {
       for (const auto& s : stack.template get< tag::param, eq, tag::bcdir >())
         if (s.empty())
           Message< Stack, ERROR, MsgKey::BC_EMPTY >( stack, in );
+
+      // Put in default farfield pressure if not specified by user
+      // if outlet BC is configured for this compflow system
+      auto& bcoutlet = stack.template get< tag::param, eq, tag::bcoutlet >();
+      if (!bcoutlet.empty() || bcoutlet.size() != neq.get< eq >()) {
+        auto& fp =
+          stack.template get< tag::param, eq, tag::farfield_pressure >();
+        if (fp.size() != bcoutlet.size()) fp.push_back( 1.0 );
+      }
```

Comment:
This block is executed when a `compflow ... end` input file block has finished parsing. What we do here is to set the default farfield pressure. The logic is that if outlet bc is specified, there should be a farfield pressure as well. If this is not given, we set it to 1.0, if given, the given value is used.

We set the defaults this way because the farfield pressure value is not stored inside the `CompFlowPDEParameters` tuple in `Control/Inciter/Types.hpp`. Its type is a `std::vector` because there can be potentially multiple `compflow ... end` blocks in a single input file and we need to be able to address each individually. This makes not much sense for a compflow block, but compflow is just another PDE system type in the parser/grammar, which by design allows multiple coupled PDE systems, e.g., flow (governing conservation of mass momentum and energy) + combustion (governing scalars only). This is why the logic says
```
      if (!bcoutlet.empty() || bcoutlet.size() != neq.get< eq >()) {
```
which detects if outlet bc is configured by the user at all for this compflow block (not empty) or if this is not the first one (and the previous one(s) potentially have outlet bcs), this most recently parsed compflow block does not have outlet bc. Then inside this if block, we detect if the farfield pressure is given. If not, we assign it.